Sage/carbon
# refactor(global-style): convert file to typescript FE-4352

### Proposed behaviour


Refactors `style` directory files to TypeScript

### Current behaviour


The `style` directory files are JavaScript

### Checklist




#### QA


### Additional context



### Testing instructions


Files changed relate to theme and some style utils so should just be covered by chromatic and only require manual testing as an additional check for things like the spacing props (the filter utils have also been refactored)

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
